### PR TITLE
Fixed invalid hex number type association from number to blob

### PR DESCRIPTION
**Pull request by Hydrocharged**
Fixes https://github.com/dolthub/dolt/issues/2968

### CODE REVIEW

**Comment by milahu** [comment]
off topic

the `Eval` function is a little obscure ... what's the point of doing `round(x * 10^d) / 10^d`?

```go
// Eval implements the Expression interface.
func (r *Round) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
// ...
	case float64:
		return math.Round(xNum*math.Pow(10.0, dVal)) / math.Pow(10.0, dVal), nil
```

can we add a comment to explain what is `x` and `d` = left and right

d = numDecimalPlaces as in format.go?

File: enginetest/script_queries.go [hunk 1]
```diff
@@ -1516,6 +1516,23 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "Handle hex number to binary conversion",
+		SetUpScript: []string{
+			"CREATE TABLE hex_nums (pk BIGINT PRIMARY KEY, v1 INT, v2 BIGINT UNSIGNED, v3 DOUBLE, v4 BINARY(32));",
```
Comment (zachmu) [inline enginetest/script_queries.go:1522]: Put a varbinary and a blob in there too for good measure

**Review by zachmu — APPROVED** [review]
LGTM

**Comment by Hydrocharged** [comment]
@milahu I've added some comments per your suggestion, but just to go over it here:

One way to implement rounding to a decimal place is to shift the number up by the desired decimal position, round to the nearest integer (allows us to use the built-in rounding function), and then shift down by the same number of decimals.

For example, let's say we have `5.855` and want to round to 2 decimal places, which should give us `5.86`. In this example, that would mean `x = 5.855` and `d = 2`
* `round(x * 10^d) / 10^d` This is the initial rounding function
* `round(5.855 * 10^2) / 10^2` We replace the variables with numbers
* `round(5.855 * 100) / 100` Calculate the exponents
* `round(585.5) / 100` Do the inner multiplication before the round, which shifts our number up by 2 decimal places
* `586 / 100` Perform the rounding
* `5.86` Finally do the division, which shifts our number back down by 2 decimal places

**Comment by milahu** [comment]
aah, of course. facepalm.
so `d` is the rounding precision

i guess that float-formatting like `%.2f` cannot be used here (in golang),
because it does not round, but cut off after x digits
so `format(0.999, '%.2f')` would be `0.99` and not `1.00`

in python, this would work: `"{:.2f}".format(0.999) == '1.00'`

**Pull request merged**